WordPressEditor/WordPressEditor/Classes/Processors/EmbedURLProcessor.swift

```diff
@@ -49,6 +57,74 @@ public struct EmbedURLProcessor{
         return matches(regex) || matches(channel) || matches(player)
     }
 
+    /// Tests the url to see if it's a valid Twitter URL.
+    ///
+    /// Supports these formats:
+    ///  - Tweet
+    ///  - Profile
+    ///  - Likes
+    ///  - List
+    ///  - Moment
+
+    public var isTwitterEmbed: Bool {
+
+        return matchesAnyOf([
+            pattern("https?://(www\\.)?twitter\\.com/\\w{1,15}/status(es)?/[\\S]+$"),   // Status
+            pattern("https?://(www\\.)?twitter\\.com/\\w{1,15}$"),                      // Profile
+            pattern("https?://(www\\.)?twitter\\.com/\\w{1,15}/likes$"),                // Likes
+            pattern("https?://(www\\.)?twitter\\.com/\\w{1,15}/lists/[\\S]+$"),         // List
+            pattern("https?://(www\\.)?twitter\\.com/i/moments/[\\S]+$"),               // Moments
+        ])
+    }
+
+    /// Tests the url to see if it's a valid Flickr URL.
+    ///
+    /// Supports these formats:
+    ///  - Photo URL
+    ///  - Gallery URL
+    ///  - Flic.kr short URLs (photos and galleries)
+    ///
+    public var isFlickrEmbed: Bool {
+        let regex = pattern("^https?://(www.)?(flickr.com|flic\\.kr)/(photos|p|s)/[\\S]+$")
+        return matches(regex)
+    }
+
+    /// Tests the url to see if it's a valid Issuu URL.
+    public var isIssuuEmbed: Bool {
+        return matches(pattern("^https?://(www\\.)?issuu\\.com/.+/docs/[\\S]+$"))
+    }
+
+    /// Tests the url to see if it's a valid Instagram URL.
+    public var isInstagramEmbed: Bool {
+        return matches(pattern("^https?://(www\\.)?instagr(\\.am|am\\.com)/p/[\\S]+$"))
+    }
+
+
+    /// Tests the url to see if it's a valid Facebook URL.
+    ///
+    /// Supports these formats:
+    ///  - Post URL
+    ///  - Photo URL
+    ///  - Note URL
+    ///  - Video URL
+    ///
+    public var isFacebookEmbed: Bool {
+        return matchesAnyOf([
+            pattern("https?://www\\.facebook\\.com/.*/posts/[\\S]+$"),                    // Post
+            pattern("https?://www\\.facebook\\.com/.*/photos/[\\S]+$"),                   // Photo
+            pattern("https?://www\\.facebook\\.com/notes/[\\S]+$"),                       // Note
+            pattern("https?://www\\.facebook\\.com/.*/videos/[\\S]+$"),                   // Video
+        ])
+    }
+
+    private func pattern(_ pattern: String) -> NSRegularExpression {
+        return try! NSRegularExpression(pattern: pattern, options: [.caseInsensitive])
+    }
+
+    private func matchesAnyOf(_ regexes: [NSRegularExpression]) -> Bool{
```

Comment:
IMHO a more "Swifty" name for this method would be `matches(anyOf:)`, since the "matches" part refers to the nature of the method, and "any of" refers specifically to the first input parameter.

Reply:
Nice! That's a way better way to do it :) 

Updated!